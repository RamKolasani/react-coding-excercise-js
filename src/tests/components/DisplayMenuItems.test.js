import React from 'react';
import { shallow } from 'enzyme';
import DisplayMenuItems from '../../components/DisplayItems';

const item={
    "name": "Coffee",
    "category": "Beverages",
    "itemId": 16306,
    "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/Jw5kV6JUEeqSXYyFkDJJSQ.jpg",
    "price": 200,
    "tax": 8.25,
    "available": true,
    "description": "",
    "modifierGroups": {},
    "timePeriods": [
      {
        "price": null,
        "description": null,
        "daysOfWeek": [],
        "period": [],
        "alwaysAvailable": true
      }
    ]
  };

describe('test DisplayMenuItems component',()=>{
    let wrapper;
    beforeEach(()=>{
         wrapper = shallow(<DisplayMenuItems  item={item} />);
    })
    test('test DisplayMenuItem snapshot', ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test('should render image', ()=>{
        expect(wrapper.find('img')).toHaveLength(1);
    })

    test('should render item details', ()=>{
        expect(wrapper.find('p')).toHaveLength(5);
    })
})