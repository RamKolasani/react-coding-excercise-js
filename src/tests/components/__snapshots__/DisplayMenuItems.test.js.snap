// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DisplayMenuItems component test DisplayMenuItem snapshot 1`] = `
<Link
  to={
    Object {
      "pathname": "/view-item/16306",
      "state": Object {
        "available": true,
        "category": "Beverages",
        "description": "",
        "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/Jw5kV6JUEeqSXYyFkDJJSQ.jpg",
        "itemId": 16306,
        "modifierGroups": Object {},
        "name": "Coffee",
        "price": 200,
        "tax": 8.25,
        "timePeriods": Array [
          Object {
            "alwaysAvailable": true,
            "daysOfWeek": Array [],
            "description": null,
            "period": Array [],
            "price": null,
          },
        ],
      },
    }
  }
>
  <div
    className="display-items"
  >
    <img
      alt="Coffee image"
      className="item-image"
      src="https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/Jw5kV6JUEeqSXYyFkDJJSQ.jpg"
    />
    <div
      className="item-details"
    >
      <h3
        className="item-name"
      >
        Coffee
      </h3>
      <ul>
        <li>
          <p>
            <strong>
              Description: 
            </strong>
          </p>
        </li>
        <li>
          <p>
            <strong>
              Category: 
            </strong>
            Beverages
          </p>
        </li>
        <li>
          <p>
            <strong>
              Item Id: 
            </strong>
            16306
          </p>
        </li>
        <li>
          <p>
            <strong>
              Availability: 
            </strong>
            Available
          </p>
        </li>
      </ul>
      <p
        className="price"
      >
        <strong>
          Price: ₹
          200
           | Tax: ₹
          8.25
        </strong>
      </p>
    </div>
  </div>
</Link>
`;
